cdap-distributions/src/etc/cdap/conf.dist/cdap-site.xml

```diff
@@ -20,5 +20,21 @@
     Your site level configuration goes here 
   -->
 
+  <property>
+    <name>dataset.service.exec.threads</name>
+    <value>300</value>
```

Comment:
It's what we recommended during several installs, and what was found to be optimal for performance.
Also, see the main description in https://issues.cask.co/browse/CDAP-4840, which mentions the same number.